dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserController.java

```diff
@@ -653,37 +656,30 @@ public WebMessage putJsonObject( @PathVariable( "uid" ) String pvUid, @CurrentUs
         HttpServletRequest request )
         throws IOException,
         ConflictException,
-        ForbiddenException
+        ForbiddenException,
+        NotFoundException
     {
         User inputUser = renderService.fromJson( request.getInputStream(), getEntityClass() );
 
-        List<User> users = getEntity( pvUid, NO_WEB_OPTIONS );
-        if ( users.isEmpty() )
-        {
-            throw new ConflictException( getEntityName() + " does not exist: " + pvUid );
-        }
+        User user = getEntity( pvUid );
 
         // TODO: To remove when we remove old UserCredentials compatibility
-        populateUserCredentialsDtoCopyOnlyChanges( users.get( 0 ), inputUser );
+        populateUserCredentialsDtoCopyOnlyChanges( user, inputUser );
 
         return importReport( updateUser( pvUid, inputUser ) )
             .withPlainResponseBefore( DhisApiVersion.V38 );
     }
 
     protected ImportReport updateUser( String userUid, User inputUser )
         throws ConflictException,
-        ForbiddenException
+        ForbiddenException,
+        NotFoundException
     {
-        List<User> users = getEntity( userUid, NO_WEB_OPTIONS );
-
-        if ( users.isEmpty() )
-        {
-            throw new ConflictException( getEntityName() + " does not exist: " + userUid );
```

Comment:
FYI: This is technically a change in API behaviour. Previously an update attempt on a non existing user would result in conflict whereas it now results in not found. Not found is consistent with any other update.